### PR TITLE
86cutf5yg/renew token

### PR DESCRIPTION
**Pull request by abhijith-biju**
## Overview
This pull request introduces integration of renew-token and logout api.

### 1. Renew Token funcitonality
- Renew JWT tokens periodically before access token expires.
- Add feature to retry renew-token once more on failure(excluding 401 error).
- Redirect to login modal on 401 responses.

### 2. Logout functionality
- Integrated logout api to revoke tokens for improved safety.

### 3. UI Changes
- Add confirmation alert when closing sign-up modal after making any changes.
- Remove auto-capitalization for button texts. 

## UI Reference
confirmation alert

### CODE REVIEW

**Review by akshay-krishna — APPROVED** [review]

File: src/utils/authUtils/authUtils.ts [hunk 1]
```diff
@@ -1,28 +1,112 @@
...
+};
+
+// This function creates a setInterval to renew token periodically
+// and returns the intervalId
+// returns undefined if remainingTime for expiry cannot be found
+export const setRefreshInterval = (accessToken: string) => {
+    const remainingTime = getRemainingTimeForExpiry(accessToken);
+    if (remainingTime) {
+        // renew token every 2 minutes before access token expires
+        const intervalForRenew = remainingTime - 120000;
```
Comment (sania-salim) [inline src/utils/authUtils/authUtils.ts:107]: What happens when negative intervalForRenew appears?

File: src/hooks/useLogout.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                navigate(paths.home);
+                dispatch(logout());
+                clearAuthDataFromStorage();
+                toast.success(t('logoutSuccessMessage'), {
+                    toastId: 'logout success',
+                });
+                return;
+            }
+
+            console.error(error);
```
Comment (sania-salim) [inline src/hooks/useLogout.ts]: Do we need to log error to console?

**Review by sania-salim — APPROVED** [review]

File: src/hooks/useLogout.ts (hunk 2 above)
Comment (abhijith-biju) [reply src/hooks/useLogout.ts]: Removed.

File: src/utils/authUtils/authUtils.ts (hunk 1 above)
Comment (abhijith-biju) [reply src/utils/authUtils/authUtils.ts:107]: I updated to set a minimum interval of 1 minute.

**Pull request merged**